File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/roleplay/ObjectItemInRolePlay.scala





















// Generated on 05/03/2013 10:35:05
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait ObjectItemInRolePlay extends TypeDeserializer {
  val typeId = 198

  protected trait Type extends BaseType with Serializable {
    val typeId = ObjectItemInRolePlay.typeId

    val cellId: Short
    val objectGID: Short

    def serialize(buf: Buffer) {
      buf.writeShort(cellId)
      buf.writeShort(objectGID)
    }

  }

  protected def create(cellId: Short, objectGID: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val cellId = buf.readShort()
    val objectGID = buf.readShort()

    create(cellId, objectGID)
  }

}

object ObjectItemInRolePlay extends ObjectItemInRolePlay {
  case class TypeImpl(cellId: Short, objectGID: Short) extends Type
  override def create(cellId: Short, objectGID: Short) = TypeImpl(cellId, objectGID)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(cellId: Short, objectGID: Short) = create(cellId, objectGID)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/ObjectItemInRolePlay.java


// Generated on 05/08/2013 19:38:01
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ObjectItemInRolePlay extends NetworkType {
    public static final short TYPE_ID = 198;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short cellId;
    public short objectGID;
    
    public ObjectItemInRolePlay() { }
    
    public ObjectItemInRolePlay(short cellId, short objectGID) {
        this.cellId = cellId;
        this.objectGID = objectGID;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(cellId);
        buf.writeShort(objectGID);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        cellId = buf.readShort();
        if (cellId < 0 || cellId > 559)
            throw new RuntimeException("Forbidden value on cellId = " + cellId + ", it doesn't respect the following condition : cellId < 0 || cellId > 559");
        objectGID = buf.readShort();
        if (objectGID < 0)
            throw new RuntimeException("Forbidden value on objectGID = " + objectGID + ", it doesn't respect the following condition : objectGID < 0");
    }
    
}
